lib/context_adapter_helper.js

```diff
@@ -516,43 +614,50 @@ function getServiceDescriptor(request, operationDescriptor, callback) {
  * @return {object} The payload to be sent in the request to a Third Party
  */
 function getThirdPartyPayload(serviceDescriptor, operationDescriptor) {
-  var payload;
-  if (operationDescriptor.action === caConfig.BUTTON_ENTITY.OPERATION_ACTION.SYNCHRONOUS) {
-    // TODO For the time being we return the operation descriptor itself
-    return operationDescriptor;
-  } else if (operationDescriptor.action === caConfig.BUTTON_ENTITY.OPERATION_ACTION.ASYNCHRONOUS_CREATE) {
-    payload = {
-      operationDescriptor: operationDescriptor,
-      webHook: 'http://' + caConfig.CA_HOST + ':' + caConfig.CA_PORT +
-      caConfig.CA_PATH + caConfig.UPDATE_WEB_HOOK_PATH
-    };
-    return payload;
+  var payload = {
+    button: operationDescriptor.buttonId,
+    action: operationDescriptor.action,
+    extra: operationDescriptor.extra
+  };
+  if (serviceDescriptor.interactionType ===
+      caConfig.SERVICE_ENTITY.INTERACTION_TYPES.ASYNCHRONOUS) {
+    payload.callback = 'http://' + caConfig.CA_HOST + ':' + caConfig.CA_PORT +
+      caConfig.CA_PATH + caConfig.CA_CALLBACK_PATH;
   }
+  return payload;
 }
 
 /**
  * Generates the result in case of error or from the response obtained from the third party
  * @param {object} err The error, if any
+ * @param {object} data The data returned by the Third Party as the operation request result
  * @return {object} The generated operation result
  */
-function getResult(err) {
+function getResult(err, data) {
   if (err) {
     // TODO For the time being we return '0,<error code>,<details>'
     return '0,' + err.code + ',' + err.message;
-  } else {
-    // TODO For the time being we return '1' as the result
-    return '1';
   }
-}
 
-/**
- * Returns the external request id from the response received from the third party
- * @param {object} response The response received from the third party
- * @return {object} The external request id
- */
-function getExternalRequestId(response) {
-  // TODO For the time being we return the response received from the third party
-  return response;
+  var detailsObj;
+  var result = '';
+  if (data) {
+    if (data.body) {
+      detailsObj = data.body.details;
+    } else if (data.payload) {
+      detailsObj = data.payload.details;
+    } else {
+      return '0,' + 'XXX' + ',' + 'Unknown data received from the Third Party as the request result';
```

Comment:
Fixed in https://github.com/telefonicaid/context-adapter/commit/bb9e16a6164ba8bf30505511ef9e0568fc56812f